webapi/src/main/scala/org/knora/webapi/responders/admin/UsersResponderADM.scala

```diff
@@ -1420,34 +1403,50 @@ class UsersResponderADM(responderData: ResponderData) extends Responder(responde
       * Helper method for checking if an username is already registered.
       *
       * @param username the username of the user.
+      * @param current the current username of the user.
       * @return a [[Boolean]].
       */
-    private def userByUsernameExists(username: String): Future[Boolean] = {
-        for {
-            askString <- Future(queries.sparql.admin.txt.checkUserExistsByUsername(username = username).toString)
-            // _ = log.debug("userExists - query: {}", askString)
+    private def userByUsernameExists(username: Option[String], current: Option[String] = None): Future[Boolean] = {
+        if (!username.isDefined) {
+            FastFuture.successful(false)
+        } else if (current.isDefined && username.get.equals(current.get)) {
+            FastFuture.successful(true)
+        } else {
+            stringFormatter.validateUsername(username.get, throw BadRequestException(s"The username: '${username.get}' contains invalid characters"))
+            for {
+                askString <- Future(queries.sparql.admin.txt.checkUserExistsByUsername(username = username.get).toString)
+                // _ = log.debug("userExists - query: {}", askString)
 
-            checkUserExistsResponse <- (storeManager ? SparqlAskRequest(askString)).mapTo[SparqlAskResponse]
-            result = checkUserExistsResponse.result
+                checkUserExistsResponse <- (storeManager ? SparqlAskRequest(askString)).mapTo[SparqlAskResponse]
+                result = checkUserExistsResponse.result
 
-        } yield result
+            } yield result
+        }
     }
 
     /**
       * Helper method for checking if an email is already registered.
       *
       * @param email the email of the user.
+      * @param current the current email of the user.
       * @return a [[Boolean]].
       */
-    private def userByEmailExists(email: String): Future[Boolean] = {
-        for {
-            askString <- Future(queries.sparql.admin.txt.checkUserExistsByEmail(email = email).toString)
-            // _ = log.debug("userExists - query: {}", askString)
+    private def userByEmailExists(email: Option[String], current: Option[String] = None): Future[Boolean] = {
```

Comment:
Why use an `Option` for `email`? Is there a reason to call this function if you don't have a new email address?